power_grid_model_c/power_grid_model/include/power_grid_model/container.hpp

```diff
@@ -319,12 +319,16 @@ class Container<RetrievableTypes<GettableTypes...>, StorageableTypes...> {
       public:
         explicit Proxy(container_type& container)
             : begin_{&container, 0}, end_{&container, container.template size<base_type>()} {}
-        Iterator<Gettable> begin() { return begin_; }
-        Iterator<Gettable> end() { return end_; }
+
+        auto begin() -> Iterator<Gettable> { return begin_; }
```

Comment:
Why do we need trailing return types here?

Reply:
we don't necessarily need it but it keeps the function names lined up